account_custom_account_by_invoice_address/models/account_move_line.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2025 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = "account.move.line"
+
+    def _compute_account_id(self):
+        res = super()._compute_account_id()
+        for rec in self.filtered(lambda r: r.display_type == "payment_term"):
+            move = rec.move_id
+            if not move.partner_id.parent_use_invoice_address_accounts:
+                continue
```

Comment:
When you use filtered as input of your loop, you create a double loop. A first time into filtered and a second time on the recordset return by filtered. It's more efficient to test the display_type directly into a loop on self....
```suggestion
        for rec in self:
            if rec.display_type != "payment_term":
                continue
            move = rec.move_id
            if not move.partner_id.parent_use_invoice_address_accounts:
                continue
```